ts/main.ts

```diff
@@ -28,9 +37,93 @@ document.addEventListener('DOMContentLoaded', () => {
     data.entries.unshift(newEntry);
     data.nextEntryId++;
 
+    const $entry = renderEntry(newEntry);
+    $entriesList.prepend($entry);
+
     $photoPreview.setAttribute('src', 'images/placeholder-image-square.jpg');
     $form.reset();
 
     savedData();
+    toggleNoEntries();
+    viewSwap('entries');
+  });
+
+  data.entries.forEach((entry) => {
+    const $entry = renderEntry(entry);
+    $entriesList.appendChild($entry);
   });
+
+  toggleNoEntries();
+  viewSwap(data.view);
+
+  $newEntryButton.addEventListener('click', (event) => {
+    event.preventDefault();
+    viewSwap('entry-form');
+  });
+
+  if ($entriesLink) {
+    $entriesLink.addEventListener('click', (event) => {
+      event.preventDefault();
+      viewSwap('entries');
+    });
+  }
 });
+
+function renderEntry(entry: JournalEntry): HTMLElement {
+  const $entryItem = document.createElement('li');
+  $entryItem.className = 'entry-item';
+
+  const $entryImage = document.createElement('div');
+  $entryImage.className = 'entry-image';
+  const $img = document.createElement('img');
+  $img.setAttribute('src', entry.photoUrl);
+  $img.setAttribute('alt', 'Entry Image');
+  $entryImage.appendChild($img);
+
+  const $entryContent = document.createElement('div');
+  $entryContent.className = 'entry-content';
+  const $title = document.createElement('h3');
+  $title.textContent = entry.title;
+  const $notes = document.createElement('p');
+  $notes.textContent = entry.notes;
+  $entryContent.appendChild($title);
+  $entryContent.appendChild($notes);
+
+  $entryItem.appendChild($entryImage);
+  $entryItem.appendChild($entryContent);
+
+  return $entryItem;
+}
+
+function toggleNoEntries(): void {
+  const $entriesList = document.getElementById(
+    'entries-list',
+  ) as HTMLUListElement;
+  const $noEntries = document.getElementById('no-entries') as HTMLElement;
+  if (data.entries.length === 0) {
+    $noEntries.classList.remove('hidden');
+    $entriesList.classList.add('hidden');
+  } else {
+    $noEntries.classList.add('hidden');
+    $entriesList.classList.remove('hidden');
+  }
+}
+
+function viewSwap(view: string): void {
+  const $entryFormView = document.querySelector(
+    '[data-view="entry-form"]',
+  ) as HTMLElement;
+  const $entriesView = document.querySelector(
+    '[data-view="entries"]',
+  ) as HTMLElement;
```

Comment:
Same thing with these, these are available on the page when the HTML loads so these queries can be moved to the top of the file